rotkehlchen/db/snapshots.py

```diff
@@ -223,41 +223,57 @@ def _export(
     ) -> Tuple[bool, str]:
-        """
-        Converts the snapshot files to list to dictionaries.
-        Performs a series of validation checks on the list before importing.
-        """
-        balances_list = self._csv_to_dict(balances_snapshot_file)
-        location_data_list = self._csv_to_dict(location_data_snapshot_file)
+        """Validates the snapshot data about to be imported."""
         # check if the headers match the type stored in the db
         has_invalid_headers = (
-            tuple(balances_list[0].keys()) != ('timestamp', 'category', 'asset_identifier', 'amount', 'usd_value') or  # noqa: E501
-            tuple(location_data_list[0].keys()) != ('timestamp', 'location', 'usd_value')  # noqa: E501
+            tuple(balances_data[0].keys()) != ('timestamp', 'category', 'asset_identifier', 'amount', 'usd_value') or  # noqa: E501
+            tuple(location_data[0].keys()) != ('timestamp', 'location', 'usd_value')  # noqa: E501
         )
         if has_invalid_headers:
-            return False, 'csv file has invalid headers'
+            return False, f'{source} has invalid headers'
 
         # check if all timestamps are the same.
-        balances_timestamps = [int(entry['timestamp']) for entry in balances_list]
-        location_data_timestamps = [int(entry['timestamp']) for entry in location_data_list]
+        balances_timestamps = [int(entry['timestamp']) for entry in balances_data]
+        location_data_timestamps = [int(entry['timestamp']) for entry in location_data]
         has_different_timestamps = (
             balances_timestamps.count(balances_timestamps[0]) != len(balances_timestamps) or
             location_data_timestamps.count(location_data_timestamps[0]) != len(location_data_timestamps) or  # noqa: E501
             balances_timestamps[0] != location_data_timestamps[0]
         )
         if has_different_timestamps:
-            return False, 'csv file has different timestamps'
+            return False, f'{source} has different timestamps'
 
         # check if the timestamp can be converted to int
         try:
-            _ = deserialize_timestamp(balances_list[0]['timestamp'])
+            _ = deserialize_timestamp(balances_data[0]['timestamp'])
         except DeserializationError:
-            return False, 'csv file contains invalid timestamp format'
+            return False, f'{source} contains invalid timestamp format'
+
+        return True, ''
+
+    def import_snapshot(
+        self,
+        balances_snapshot_file: Path,
+        location_data_snapshot_file: Path,
+    ) -> Tuple[bool, str]:
+        """
+        Converts the snapshot files to list to dictionaries.
+        Performs a series of validation checks on the list before importing.
+        """
+        balances_list = self._csv_to_dict(balances_snapshot_file)
+        location_data_list = self._csv_to_dict(location_data_snapshot_file)
+        is_valid, message = self.validate_import_data(
```

Comment:
I might be wrong though but the `snapshots.py` module encapsulates the entirety of what's needed to import, export, delete & edit a snapshot. That means all methods particular to the snapshot manipulation. 

Due to that, only showing the needed method to the outside world i.e `.export(), import(), delete()` 
Once again, it might just be me overthinking.

Your thoughts?

Reply:
Hmhmhm. You think that `snapshots.py` show to the outside world `.export()`, `.import()` and `.delete()`, right? If yes, I agree. My point is about that you we should better pass `location_data` and `balances_data` (data that is fully prepared to be written to the db) rather that some extra sources such as `Path`

Reply:
Back to this...

`import_data()` handles the data preparation and validation while `_import_data()` does the actual importing.

If I move the preparation out of `import_data()`, then I'll still have to call a method of `DBSnapshot` to handle the preparation inside `rest.py`. This is because the validation is snapshot-specific i.e it's not a generic validation per say that can be used elsewhere.

Reply:
Got you. One last question :smile: Will it work if you read and validate all data in `rest.py` and pass to the database `balances_list` and `location_data_list` with this prepared data?

Reply:
Sure sure. It'll. I'd just do that ;)